### PR TITLE
Sin/playlists

### PR DESCRIPTION
**Pull request by Zindre17**

### CODE REVIEW

File: api/crunch.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                    })
+                }
+            }
+        }
+    });
+}
+
+function fetchPlayedTracksLast7days(): Promise<PlayedTrack[]> {
+    return prisma.$queryRaw<PlayedTrack[]>`
+    SELECT
```
Comment (adriansberg) [inline api/crunch.ts:144]: No way to achieve this without a raw query?

File: api/crunch.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            and t.genre is not null)
+    GROUP BY 
+          g.id
+        , u.id
+        , t.id
+        , t.genre
+ `;
+}
+
+(BigInt.prototype as any).toJSON = function () {
```
Comment (adriansberg) [inline api/crunch.ts:167]: Where's this used?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -66,3 +69,60 @@ model track_artists {
 
   @@id([track_id, artist_id], map: "primary")
 }
+
+/// This model or at least one of its fields has comments in the database, and requires an additional setup for migrations: Read more: https://pris.ly/d/database-comments
```
Comment (adriansberg) [inline prisma/schema.prisma:76]: What do these comments mean?

File: api/crunch.ts (hunk 1 above)
Comment (Zindre17) [reply api/crunch.ts:144]: Maybe, but I don't want to spend more time on wrestling with an ORM when I know how to do it without one :D

File: api/crunch.ts (hunk 2 above)
Comment (Zindre17) [reply api/crunch.ts:167]: I think all of the number from the database come as BigInt, and they have no default json serialization.

File: prisma/schema.prisma (hunk 3 above)
Comment (Zindre17) [reply prisma/schema.prisma:76]: No idea :upside_down_face:

**Review by adriansberg — APPROVED** [review]
Great! Set up cron job for this as well?

**Pull request merged**